### ISSUE TITLE
status (SAT/UNSAT) should not be set from exit code, but from contents of answer?

### ISSUE THREAD

**Issue by jwaldmann** [ekmett/ersatz#96]
I am trying to use solver PRS (https://github.com/shaowei-cai-group/PRS-sc24/) .

Their exit code is 0 always.
```
$ /opt/PRS/sc24/bin/PRS check.cnf 
s SATISFIABLE
v -1 2 0
$ echo $?
0

$ /opt/PRS/sc24/bin/PRS uns.cnf
s UNSATISFIABLE
$ echo $?
0
```

Ersatz currently has
```
...
    (exit, out, _err) <-
      readProcessWithExitCode path [problemPath] options
    let sol = parseSolution5 out
    return (resultOf exit, sol)

resultOf :: ExitCode -> Result
resultOf (ExitFailure 10) = Satisfied
resultOf (ExitFailure 20) = Unsatisfied
resultOf _                = Unsolved
```
This is redundant, the status also is in the output, and we have to parse it anyway, to get the assignment.

I have some code for that, in a different project https://git.imn.htwk-leipzig.de/waldmann/pure-matchbox/-/blob/master/src/Ersatz/Solver/Satcomp.hs?ref_type=heads that I should extract and present as a pull request here.

**Comment by RyanGlScott** [comment]
Indeed, I suspect that this works for most QBF solvers due to an exit code requirement imposed by various QBF competitions (e.g., [QBFEVAL](https://www.qbflib.org/qbfeval20.php)). I can certainly believe that not all solvers adhere to this convention.

That being said, I'm not sure what exactly you are proposing. Each solver already has specific treatment in `ersatz` (e.g., see the [code for handling Z3](https://github.com/ekmett/ersatz/blob/333858e587e9a34e9135200bccf41356eab14a3b/src/Ersatz/Solver/Z3.hs), which does not use `resultOf`). Couldn't we just add a specific case for PRS that avoids the use of `resultOf`, similarly to how Z3 is handled?

**Comment by RyanGlScott** [comment]
That makes sense! I would welcome making this code more standardized. Of course, we should make sure to test each solver to make sure that it continues to work with the new approach.
